### DISCUSSION TITLE
Exploring Support for --reinstall-packages-from=current and --latest-npm Flags in nvm for Windows: An Inquiry and Alternative Solutions

### DISCUSSION THREAD

**Discussion by issam-seghir** [coreybutler/nvm-windows#1019] [Support] [answered]
Is the nvm for Windows currently supporting or planning to support the `--reinstall-packages-from=current` and `--latest-npm` flags in the future?
If not, is there any alternative way to do that ( migrating my global packages and updating npm to the latest version) ?

**Comment by coreybutler** [comment, marked as answer]
No. NVM4W is not the same as nvm and neither project has a goal of being in sync with each other.

That said, we are working on a feature like this for [Runtime](https://github.com/coreybutler/nvm-windows/wiki/Runtime) right now. While the final solution may change, we are currently planning to maintain a list of global modules with the option to auto-install them when new node versions are installed. With npm, we are also working on the ability to specify a preference to choose which version of npm to install (i.e. default, latest, etc). Runtime is also slated to have an optional feature to auto-update existing versions of npm when they become available.

Runtime has significant new features over traditional version managers (like a UI, among others), so there are newer/better options for keeping the state of Node in sync with what users prefer. In other words, the challenges are being addressed, but in a different manner.

In the meantime, updating npm is as simple as `npm i -g npm`. Migrating global packages doesn't have a comparable solution, but you could write a script to do this if you really need to.

**Reply by issam-seghir** [reply]
Great ! I want to express my gratitude for your dedication to bringing this wonderful tool to Windows.
I'm genuinely thrilled about the promising future it holds

**Reply by zxanderh** [reply]
I too am grateful for NVM4W and look forward to Runtime! In the meantime, I decided to take @coreybutler's suggestion and write a script. I figured I'd post a link in case someone else stumbles upon this thread in the future: [https://gist.github.com/zxanderh/28e39fd289227af6431c085c13ac37a3](https://gist.github.com/zxanderh/28e39fd289227af6431c085c13ac37a3)
